.github/workflows/test-python.yml

```diff
@@ -0,0 +1,50 @@
+---
+name: test-python
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - main
+      - renovate/**
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "pypy3.8"
+          - "pypy3.9"
+          - "pypy3.10"
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
```

Comment:
While our CI is free, it is not unlimited and we are frequently running out of runners. So I think the current matrix is a bit excessive. 

Can we reduce this to all currently supported python versions on `ubuntu-latest` and then only the latest python version on `windows-latest` and `macos-latest`?